torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -292,19 +294,20 @@ void GPULower::replaceSizes() {
   // Set domains to be based on symbolic sizes (i.e. Ti.size[...])
   for (TensorView* tv : all_tvs) {
     std::vector<IterDomain*> new_domain_iters;
-    TensorDomain* root_td = tv->getRootDomain();
+    const std::vector<IterDomain*>& root_td = tv->getRootDomain();
 
-    for (decltype(root_td->nDims()) i{0}; i < root_td->nDims(); i++) {
-      Val* new_size = root_td->axis(i)->extent();
+    for (decltype(root_td.size()) i{0}; i < root_td.size(); i++) {
```

Comment:
ditto

Reply:
Made a note on https://github.com/csarofeen/pytorch/issues/34 to run through and clean decltypes up.